rust/src/block/store.rs

```diff
@@ -62,4 +78,28 @@ pub trait BlockStore {
 
     /// TODO
     fn get_block_version(&self, state_hash: &BlockHash) -> anyhow::Result<Option<PcbVersion>>;
+
+    /// TODO
+    fn set_account_balance(&self, pk: &PublicKey, balance: u64) -> anyhow::Result<()>;
+
+    /// Generate account balance updates when the best tip changes
+    fn common_ancestor_account_balance_updates(
+        &self,
+        old_best_tip: &BlockHash,
+        new_best_tip: &BlockHash,
+    ) -> anyhow::Result<LedgerBalanceUpdate>;
+
+    /// TODO
+    fn set_block_balance_updates(
+        &self,
+        state_hash: &BlockHash,
+        balance_updates: Vec<PaymentDiff>,
+    ) -> anyhow::Result<()>;
+
+    /// TODO
+    fn get_block_balance_updates(&self, state_hash: &BlockHash)
+        -> anyhow::Result<Vec<PaymentDiff>>;
```

Comment:
What's with all the TODOs?

Reply:
Draft. Will leave comments

Reply:
Just wait until you hear what that `common_ancestor_account_balance_updates` is for!